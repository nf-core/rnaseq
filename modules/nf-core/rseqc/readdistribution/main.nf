process RSEQC_READDISTRIBUTION {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/rseqc:5.0.3--py39hf95cd2a_0' :
        'biocontainers/rseqc:5.0.3--py39hf95cd2a_0' }"

    input:
    tuple val(meta), path(bam)
    path  bed

    output:
    tuple val(meta), path("*.read_distribution.txt"), emit: txt
    path  "versions.yml"                            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    read_distribution.py \\
        -i $bam \\
        -r $bed \\
        > ${prefix}.read_distribution.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rseqc: \$(read_distribution.py --version | sed -e "s/read_distribution.py //g")
    END_VERSIONS
    """
}
