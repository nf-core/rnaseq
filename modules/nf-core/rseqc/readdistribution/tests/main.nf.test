nextflow_process {

    name "Test Process RSEQC_READDISTRIBUTION"
    script "../main.nf"
    process "RSEQC_READDISTRIBUTION"
    tag "modules"
    tag "modules_nfcore"
    tag "rseqc"
    tag "rseqc/readdistribution"

    test("sarscov2 paired-end [bam]") {

        when {
            process {
                """
                input[0] = [ 
                    [ id:'test', single_end: false ], // meta map
                    file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam", checkIfExists: true)
                ]
                input[1] = file(params.modules_testdata_base_path + "genomics/sarscov2/genome/bed/test.bed12", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
