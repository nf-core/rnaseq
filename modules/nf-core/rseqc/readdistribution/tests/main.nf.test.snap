{
    "sarscov2 paired-end [bam]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.read_distribution.txt:md5,56893fdc0809d968629a363551a1655f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,fc68a268ab32a0d72a66f270d6c7925e"
                ],
                "txt": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.read_distribution.txt:md5,56893fdc0809d968629a363551a1655f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fc68a268ab32a0d72a66f270d6c7925e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2023-12-14T17:17:31.66536021"
    }
}